### ISSUE TITLE
Getting error message when fetching clang-format version

### ISSUE THREAD

**Issue by NAmmann** [Sarcasm/run-clang-format#19]
When running the script I get the following error message:
`run-clang-format.py:` error: Command '[u'clang-format', '--version']' returned non-zero exit status 1`

I'm running python 2.7.5 and  LLVM 3.4.2

When removing the return statement in line [309](https://github.com/Sarcasm/run-clang-format/blob/master/run-clang-format.py#L309). The script will (of cause) continue and succeed.

**Comment by Sarcasm** [comment]
What happens if you type this command in a terminal?

Personally I have:

    $ clang-format --version
    clang-format version 9.0.1

**Comment by NAmmann** [comment]
I'm getting the following:
```
$ clang-format --version
LLVM (http://llvm.org/):
  LLVM version 3.4.2
  Optimized build.
  Built May 10 2018 (10:48:27).
  Default target: x86_64-redhat-linux-gnu
  Host CPU: core-avx2
```

**Comment by Sarcasm** [comment]
Hum, I see.
So something is wrong, but I would recommend you to add debugging, and maybe remove the `stdout=DEVNULL` argument, to see what clang-format outputs.

Also, check the return code of the `clang-format --version` if it is not 0.
Could also be related to the arguments, where one is unicode, and the other isn't:

    '[u'clang-format', '--version']'

To test if unicode is causing the issue, you can change this part:

```diff
    parser.add_argument(
        '--clang-format-executable',
        metavar='EXECUTABLE',
        help='path to the clang-format executable',
-       default='clang-format')
+       default=str('clang-format'))
```

**Comment by NAmmann** [comment]
> Also, check the return code of the `clang-format --version` if it is not 0.

The return code is indead `1` and not `0`. So this might cause the problem.

The output when removing the `stdout=DEVNULL` is the same as described above.

And changing the coding does not change anything.

**Comment by Sarcasm** [comment]
Hum, it's really an old version.
Since the run-clang-format installation recommandations for now, are to copy it in a project, I'm thinking it's okay that you patch it locally, but I'm not sure I want to upstream the change, to support such an old version.
And it looks like it's the last version before the version output was fixed (clang 3.5).

Or the `except subprocess.CalledProcessError as e` can be modified to accept an exit code of 1, although it does not feel right to me.

**Comment by Sarcasm** [comment]
I'm closing but feel free to attempt a PR if you feel like it.
